hdat/resultspec.py

```diff
@@ -25,6 +39,9 @@ def resolve_resultspecs(archive, resultspec):
     pick_by_result_id = len(resultspec_parts) == 3 and not pick_by_index
```

Comment:
It's easy to lose track of what all the resultspec parts mean in this method. Comments with example of different resultspecs on lines 35-39 would help a lot.

Reply:
Great idea! Yeah, this code is starting to get complex.  (Hopefully it is all necessary complexity)

Reply:
It would also be worth considering if we can split this code up into smaller, more readable functions?  This could help with the complexity too.